bundles/org.eclipse.ui.ide.application/src/org/eclipse/ui/internal/ide/application/IDEApplication.java

```diff
@@ -523,9 +524,12 @@ private String getHostName() {
 	 * @param workspaceUrl
 	 * @return .lock_info file.
 	 */
-	private File getLockInfoFile(URL workspaceUrl) {
-		Path lockInfoPath = Path.of(workspaceUrl.getPath(), METADATA_FOLDER, LOCK_INFO_FILENAME);
-		return lockInfoPath.toFile();
+	private static Path getLockInfoFile(URL workspaceUrl) {
+		try {
+			return Path.of(URIUtil.toURI(workspaceUrl)).resolve(LOCK_INFO_FILE);
+		} catch (URISyntaxException e) {
+			throw new IllegalArgumentException(e);
+		}
```

Comment:
Btw. this reminded me of a work that I started a while ago and that I have now made a draft in Equinox:
- https://github.com/eclipse-equinox/equinox/pull/691

When that's available this could be further simplified to (but at the moment this is not possible yet!):
```suggestion
		return URIUtil.toFilePath(workspaceUrl).resolve(LOCK_INFO_FILE);
```